pkg/controller/common/layered_node_state.go

```diff
@@ -131,16 +131,66 @@ func isNodeDone(node *corev1.Node) bool {
 	if node.Annotations == nil {
 		return false
 	}
+
+	if !isNodeConfigDone(node) {
```

Comment:
```
if !isNodeConfigDone(node) || !isNodeImageDone(node) || !IsNodeMCDState(node, daemonconsts) {
     return false
}
````

Reply:
I'd rather leave it the way that it is, only because it's pretty explicit about when we return false. Elsewhere in the code, we did something like this and I think that's how this particular bug was introduced.